### PR TITLE
ADEN-10495 | Pass Permutive keys to Appnexus and Rubicon

### PR DESCRIPTION
**Pull request by Loczi94**

### CODE REVIEW

File: src/ad-bidders/prebid/adapters/appnexus-ast.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import { context, utils } from '@ad-engine/core';
+// tslint:disable-next-line:no-blacklisted-paths
+import { permutive } from '@ad-engine/services';
```
Comment (dmnsobczak) [inline src/ad-bidders/prebid/adapters/appnexus-ast.ts:3]: Well, there is a reason why we don't want to import modules between packages. Are you sure it won't broke anything?

File: src/ad-bidders/prebid/adapters/appnexus-ast.ts [hunk 2]
```diff
@@ -48,4 +48,15 @@ export class AppnexusAst extends PrebidAdapter {
 			],
 		};
 	}
+
+	private getAdditionalKeyVals(code): object {
+		if (context.get('bidders.prebid.additionalKeyvals.appnexus')) {
+			return {
+				p_standard: permutive.getPermutiveKeys(),
```
Comment (dmnsobczak) [inline src/ad-bidders/prebid/adapters/appnexus-ast.ts:55]: Why don't you put it here?
https://github.com/Wikia/ad-engine/blob/dev/src/ad-bidders/prebid/prebid-adapter.ts#L44

File: src/ad-bidders/prebid/adapters/rubicon-display.ts [hunk 3]
```diff
@@ -46,12 +48,21 @@ export class RubiconDisplay extends PrebidAdapter {
...
 				},
 			],
 		};
 	}
+
+	private getAdditionalKeyVals(code): object {
+		if (context.get('bidders.prebid.additionalKeyvals.rubicon')) {
+			return {
+				p_standard: permutive.getPermutiveKeys(),
+				...this.getTargeting(code, this.customTargeting),
```
Comment (dmnsobczak) [inline src/ad-bidders/prebid/adapters/rubicon-display.ts:62]: We are forgetting about data from `targeting` parameter:
`this.getTargeting(code, { ...(targeting || {}), ...this.customTargeting })`

**Review by dmnsobczak — CHANGES_REQUESTED** [review]
Unit tests are failing

File: src/ad-bidders/prebid/adapters/rubicon-display.ts (hunk 3 above)
Comment (Loczi94) [reply src/ad-bidders/prebid/adapters/rubicon-display.ts:62]: Yep, you're right. 🙈

**Review by dmnsobczak — APPROVED** [review]

**Comment by dmnsobczak** [comment]
Unit tests are failing. @Loczi94 please fix :)

**Review by dmnsobczak — APPROVED** [review]

**Pull request merged**